sdk/storage/Azure.Storage.DataMovement/src/StorageResourceCheckpointDetails.cs

```diff
@@ -1,24 +1,28 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.ComponentModel;
 using System.IO;
 
 namespace Azure.Storage.DataMovement
 {
     /// <summary>
     /// Abstract class for checkpoint data related to a specific resource type.
     /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
nit: Did we check if marking a class as EB(never) also marks its properties?

Reply:
So what it does is that it hides the class. But if we don't mark the properties, if someone were to actually create the class, then it the properties would still pop up. So overkill on hiding everything

Reply:
Sounds good